src/include/distributed/multi_physical_planner.h

```diff
@@ -338,10 +338,20 @@ typedef struct Task
 	 */
 	struct TupleDestination *tupleDest;
 
+	/*
+	 * If the task was supposed to return results, so either SELECT or DML with
+	 * RETURNING. This is only set when using EXPLAIN ANALYZE.
+	 */
+	bool expectResults;
```

Comment:
Can we do without this flag?

Not a huge fan of adding another subtle (sometimes enabled?) flag to Task, which is already a super complicated (yet super important) data structure. Is it really only set for EXPLAIN ANALYZE?  That's not my reading of SetAttributeInputMetadata. What if queryCount > 1? Do we expect bytes from all queries if it is set? Can we have queries with results when the flag is not set? etc.

Changes to Task also require changes to citus_copyfuncs et al.